server.go

```diff
@@ -610,6 +671,92 @@ func (s *server) WaitForShutdown() {
 	s.wg.Wait()
 }
 
+// watchExternalIP continously checks for an updated external IP address every
+// 15 minutes. Once a new IP address has been detected, it will automatically
+// handle port forwarding rules and send updated node announcements to the
+// currently connected peers.
+//
+// NOTE: This MUST be run as a goroutine.
+func (s *server) watchExternalIP() {
```

Comment:
Big fan of this method, makes reasoning about the behavior very straightforward 👍